### ISSUE TITLE
Run integ tests against deployed test stack

### ISSUE THREAD

**Issue by jlhood** [awslabs/realworld-serverless-application#53]
Currently the backend integ tests are self-contained in that they stand up test stacks, run the integ tests against those stacks and then tear down the test stacks. This captures one type of integ testing we do in SAR, but not the other kind, which is to test against the test environment (beta) stacks deployed by the pipeline. This is causing [some confusion](https://twitter.com/forrestbrazeal/status/1184113239019397120?s=20) so we should also add a pipeline step to run integ tests against the deployed instance of the application after pipeline deployment.

**Comment by honglu** [comment]
We considered this option but since there is no more stage after the Deploy stage, even if the integ tests fail, it does not really prevent anything or rollback the deployment in Deploy stage.

**Comment by salmanap** [comment]
Couldn't we create another stage in the pipeline? run the integ tests in that stage and then promote the changes to the final stage?

**Comment by jlhood** [comment]
Yeah I think it'd be better if the pipeline had a beta and prod stack with integ tests running against the beta stack prior to deployment to the prod stack.

**Comment by honglu** [comment]
Since we are using [aws-sam-codepipeline-cd](https://github.com/awslabs/aws-sam-codepipeline-cd) to setup CD pipeline, I created an [issue](https://github.com/awslabs/aws-sam-codepipeline-cd/issues/12) there to address this issue
